### PR TITLE
Audit fixes

### PR DESCRIPTION
**Pull request by alcueca**
From here: https://hackmd.io/@devtooligan/YieldStrategyV2Review2022-12-07

### CODE REVIEW

File: contracts/Strategy.sol [hunk 1]
```diff
@@ -90,17 +90,29 @@ contract Strategy is AccessControl, ERC20Rewards, StrategyMigrator { // TODO: I'
         state = target;
     }
 
+    /// @dev State and state variable management
+    /// @param target State to transition to
+    function _transition(State target) internal {
```
Comment (devtooligan) [inline contracts/Strategy.sol:95]: 💯  clever solution for this

Comment (devtooligan) [reply contracts/Strategy.sol:95]: what about adding a final `else` to the main `_transition(State target, IPool pool_)` above to future proof it from ever getting in a bad state, that would make it a true state machine imo.

```
    } else {
          revert("Invalid state");
```

File: contracts/Strategy.sol [hunk 2]
```diff
@@ -90,17 +90,29 @@ contract Strategy is AccessControl, ERC20Rewards, StrategyMigrator { // TODO: I'
...
+    /// @notice Mock pool mint called by a strategy when trying to migrate.
+    /// @dev Will initialize the strategy and return strategy tokens.
+    /// It is expected that base has been transferred in, but no fyTokens
+    /// @return baseIn Amount of base tokens found in contract
+    /// @return fyTokenIn This is always returned as 0 since they aren't used
+    /// @return minted Amount of strategy tokens minted from base tokens which is the same as baseIn
     function mint(address, address, uint256, uint256)
         external
         override
         auth
```
Comment (devtooligan) [inline contracts/Strategy.sol:111]: it's a bit duplicitous, but we did lose the `isState(State.DEPLOYED)` check by using an internal fn here.   Prob wouldn't hurt to add it back but not a big deal either way tbh.

File: contracts/Strategy.sol [hunk 3]
```diff
@@ -112,17 +124,26 @@ contract Strategy is AccessControl, ERC20Rewards, StrategyMigrator { // TODO: I'
         isState(State.DEPLOYED)
```
Comment (devtooligan) [inline contracts/Strategy.sol:124]: I see the `isState(State.DEPLOYED)` check has been added to the internal `_init()` should we remove it here?

**Review by devtooligan — APPROVED** [review]
Changes look good.  Approved.  

As for the changes to the testing, I think it would be worth it to do a quick video walkthrough.  You've handled a lot of interesting challenges and you've developed some helpful practices that I think would be valuable to share with the team.

File: contracts/Strategy.sol (hunk 3 above)
Comment (alcueca) [reply contracts/Strategy.sol]: Removed, thanks

File: contracts/Strategy.sol (hunk 1 above)
Comment (alcueca) [reply contracts/Strategy.sol:95]: I'm using an `enum` and it is an internal function, there is no way of passing a wrong state.

**Comment by devtooligan** [comment]
I've updated the audit report to account for all of these changes.

**Review by devtooligan — APPROVED** [review]
💯

**Pull request merged**